packages/react/src/components/form/authenticating/otp.tsx

```diff
@@ -46,26 +41,24 @@ export const OTPState = ({ flowState }: Props) => {
   const { sid } = useSlashID();
   const { values, registerField, registerSubmit, setError, clearError } =
     useForm();
-  const [formState, setFormState] = useState<
-    "initial" | "input" | "submitting"
-  >("initial");
   const submitInputRef = useRef<HTMLInputElement>(null);
 
   const factor = flowState.context.config.factor;
   const hasRetried = flowState.context.attempt > 1;
   const { title, message } = getAuthenticatingMessage(factor, {
-    isSubmitting: formState === "submitting",
+    isSubmitting: flowState.hasUIState("submitting"),
     hasRetried,
   });
 
   const handleSubmit: FormEventHandler<HTMLFormElement> = useCallback(
     (e) => {
       e.preventDefault();
+      const uiState = flowState.getUiState();
+      if (!isInputState(uiState)) return;
```

Comment:
This is supposed to be done in the state machine, UI should not be worried about this

Reply:
Yes but this function kind of doubles as a type guard, so TypeScript doesn't complain about using `submit()` method. How should I adjust it then, by type casting?
```ts
(flowState as LoginState).submit(...)
```

Reply:
I guess it's better to leave it as it is then!